integration-tests/src/test/java/oracle/weblogic/kubernetes/ItExternalNodePortService.java

```diff
@@ -0,0 +1,416 @@
+            ADMIN_USERNAME_DEFAULT, ADMIN_PASSWORD_DEFAULT);
+
+    // create encryption secret
+    logger.info("Create encryption secret");
+    String encryptionSecretName = "encryptionsecret";
+    createSecretWithUsernamePassword(encryptionSecretName, domainNamespace,
+            "weblogicenc", "weblogicenc");
+
+    // Prepare the config map sparse model file from the template by replacing
+    // Public Address of the custom channel with K8S_NODEPORT_HOST
+    nextFreePort = getNextFreePort();
+    Map<String, String> configTemplateMap  = new HashMap();
+    configTemplateMap.put("INGRESS_HOST", K8S_NODEPORT_HOST);
+    configTemplateMap.put("FREE_PORT", String.valueOf(nextFreePort));
+
+    Path srcFile = Paths.get(RESOURCE_DIR,
+        "wdt-models", "nodeport.tunneling.model.template.yaml");
+    Path targetFile = assertDoesNotThrow(
+        () -> generateFileFromTemplate(srcFile.toString(),
+        TUNNELING_MODEL_FILE, configTemplateMap));
+    logger.info("Generated tunneling ConfigMap model file {0}", targetFile);
+
+    String configMapName = "jms-nodeport-tunneling-configmap";
+    List<Path> configMapFiles = new ArrayList<>();
+    configMapFiles.add(Paths.get(RESULTS_ROOT, TUNNELING_MODEL_FILE));
+
+    createConfigMapFromFiles(configMapName, configMapFiles, domainNamespace);
+
+    // create the domain CR with a pre-defined configmap
+    createDomainResource(domainUid, domainNamespace, adminSecretName,
+        OCIR_SECRET_NAME, encryptionSecretName, replicaCount, configMapName);
+
+    // wait for the domain to exist
+    logger.info("Check for domain custom resource in namespace {0}", domainNamespace);
+    withStandardRetryPolicy
+        .conditionEvaluationListener(
+            condition -> logger.info("Waiting for domain {0} to be created in namespace {1} "
+                    + "(elapsed time {2}ms, remaining time {3}ms)",
+                domainUid,
+                domainNamespace,
+                condition.getElapsedTimeInMS(),
+                condition.getRemainingTimeInMS()))
+        .until(domainExists(domainUid, DOMAIN_VERSION, domainNamespace));
+  }
+
+  /**
+   * Verify all server pods are running.
+   * Verify all k8s services for all servers are created.
+   */
+  @BeforeEach
+  public void beforeEach() {
+    logger.info("Check admin service and pod {0} is created in namespace {1}",
+        adminServerPodName, domainNamespace);
+    checkPodReadyAndServiceExists(adminServerPodName, domainUid, domainNamespace);
+    // check managed server services and pods are ready
+    for (int i = 1; i <= replicaCount; i++) {
+      logger.info("Wait for managed server services and pods are created in namespace {0}",
+          domainNamespace);
+      checkPodReadyAndServiceExists(managedServerPrefix + i, domainUid, domainNamespace);
+    }
+  }
+
+  /**
+   * The external JMS client sends 300 messages to a Uniform Distributed
+   * Queue using load balancer HTTP url which maps to custom channel on
```

Comment:
are you using LB here ?

Reply:
No, the usecase does not use LoadBalancer. Modified the description as follows ..
 ... Queue using Nodeport service http url which maps to custom channel on ...